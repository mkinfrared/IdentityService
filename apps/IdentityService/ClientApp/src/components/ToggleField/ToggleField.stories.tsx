import { Meta, Story } from "@storybook/react";

import { ToggleField } from "./ToggleField";
import css from "./ToggleField.module.scss";
import { ToggleFieldProps } from "./ToggleField.type";

export default {
  title: "COMPONENTS/ToggleField",
  component: ToggleField,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<ToggleFieldProps> = (args) => (
  <div className={css.Story}>
    <ToggleField {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
