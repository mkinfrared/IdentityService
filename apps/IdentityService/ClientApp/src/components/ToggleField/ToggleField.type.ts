import { ToggleProps } from "@identity-service/ui";

export type ToggleFieldProps = Omit<ToggleProps, "inputRef"> & {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
