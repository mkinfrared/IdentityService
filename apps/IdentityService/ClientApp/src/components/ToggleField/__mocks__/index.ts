export { default } from "./ToggleField.mock";
