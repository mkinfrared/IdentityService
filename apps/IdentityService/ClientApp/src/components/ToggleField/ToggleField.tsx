import { Toggle } from "@identity-service/ui";
import { Ref, forwardRef, memo } from "react";

import { ToggleFieldProps } from "./ToggleField.type";

const ToggleField = forwardRef(
  (props: ToggleFieldProps, ref: Ref<HTMLInputElement>) => (
    <Toggle {...props} inputRef={ref} />
  ),
);

ToggleField.displayName = "ToggleField";

export { ToggleField };

export default memo(ToggleField);
