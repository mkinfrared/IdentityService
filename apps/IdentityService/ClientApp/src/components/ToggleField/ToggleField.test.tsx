import { render } from "@testing-library/react";

import { ToggleField } from "./ToggleField";

describe("<ToggleField />", () => {
  const Component = <ToggleField />;

  it("should be defined", () => {
    expect(ToggleField).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Toggle");

    expect(element).toBeDefined();
  });
});
