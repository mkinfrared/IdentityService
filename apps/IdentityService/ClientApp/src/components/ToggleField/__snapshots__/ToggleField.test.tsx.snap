// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleField /> should match the snapshot 1`] = `
<div>
  <label
    class="Toggle-module_Toggle__l-SWp"
    data-testid="Toggle"
  >
    <input
      class="Toggle-module_input__9vPu1"
      type="checkbox"
      value=""
    />
    <div
      class="Toggle-module_container__FoqlF"
    >
      <div
        class="Toggle-module_switch__F7NhF"
      >
        <span
          class="Toggle-module_ball__bF3bD"
        />
      </div>
    </div>
  </label>
</div>
`;
